pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -44,12 +53,18 @@ func (c *clientCache) ClientConfigForVersion(version string) (*client.Config, er
 			}
 		}
 	}
+	if config, ok := c.configs[version]; ok {
+		return config, nil
+	}
 	// TODO: have a better config copy method
 	config := *c.defaultConfig
-	if len(version) != 0 {
-		config.Version = version
+	negotiatedVersion, err := client.NegotiateVersion(&config, version)
+	if err != nil {
+		return nil, err
 	}
+	config.Version = negotiatedVersion
 	client.SetKubernetesDefaults(&config)
```

Comment:
Yep, just for version defaulting. The only thing that we changed is the `Version`, so prefix, user agent etc, is still default.